src/Form/FormAnalyzer.js

```diff
@@ -291,13 +300,18 @@ class FormAnalyzer {
         this.evaluateElAttributes(this.form)
 
         // Check form contents (noisy elements are skipped with the safeUniversalSelector)
-        this.form.querySelectorAll(this.matching.cssSelector('safeUniversalSelector')).forEach(el => {
+        const formElements = this.form.querySelectorAll(this.matching.cssSelector('safeUniversalSelector'))
+        for (let i = 0; i < formElements.length; i++) {
+            // Safety cutoff to avoid huge DOMs freezing the browser
+            if (i >= 200) break
```

Comment:
Stop at the first 200 elements, so that huge forms don't cause performance issues. This was the Rumble issue.